packages/gen-ai-hub/src/utils/deployment-cache.ts

```diff
@@ -51,13 +51,29 @@ function createDeploymentCache(cache: Cache<Deployment>) {
      * @param deployments - Deployments to retrieve the IDs and models from.
      */
     setAll: (
-      opts: Omit<DeploymentResolutionOptions, 'model'>,
+      opts: DeploymentResolutionOptions,
       deployments: AiDeployment[]
     ): void => {
       // go backwards to cache the first deployment ID for each model
-      [...deployments].reverse().forEach(deployment => {
-        cache.set(getCacheKey({ ...opts, model: extractModel(deployment) }), {
-          entry: transformDeploymentForCache(deployment)
+      const cacheEntries = [...deployments]
+        .reverse()
+        .map(deployment => transformDeploymentForCache(deployment))
+        .flatMap(entry => {
+          const entries = [entry];
+          // if the entry has a model, also cache it without model and version
+          if (entry.model) {
+            entries.push({ id: entry.id });
+          }
+          // if the entry has a model version, also cache it for without model version
+          if (entry.model?.version) {
+            entries.push({ id: entry.id, model: { name: entry.model.name } });
+          }
+          return entries;
+        });
```

Comment:
This could still be shortened if we remove the checks:
```suggestion
        .flatMap(entry => [entry, { id: entry.id }, { id: entry.id, model: { name: entry.model.name } }]);
```

Even though this is a little less performant, I think I even prefer that, but no strong opinion.